### PR TITLE
scpb and scplan refactor

### PR DESCRIPTION
**Pull request by postamar**
This change contains two refactoring commits:
1. `scpb: move Node to screl, introduce scpb.TargetState`. This commit is best reviewed by looking at the changes in scpb first, and then in the protos, and then the rest. The most subsequent complexity is around the event logging and in the stage builder, which benefitted from a deeper rewrite. 
2. `scplan: make child packages internal, including scgraph and scgraphviz`. This just moves stuff around, although there are a few notable changes in `plan.go`.

### CODE REVIEW

File: pkg/jobs/jobspb/jobs.proto [hunk 1]
```diff
@@ -493,7 +493,8 @@ message TypeSchemaChangeProgress {
 
 // TypeSchemaChangeDetails is the job detail information for the new schema change job.
 message NewSchemaChangeDetails {
-  repeated cockroach.sql.schemachanger.scpb.Target targets = 1;
+
+  cockroach.sql.schemachanger.scpb.TargetState target_state = 5 [(gogoproto.nullable) = false];
```
Comment (miretskiy) [inline pkg/jobs/jobspb/jobs.proto:497]: do we need to `reserve 1,2,3`?

**Review by fqazi — APPROVED** [review]
> Reviewed 79 of 79 files at r1, 13 of 74 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74598)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @fqazi, @postamar, and @samiskin)

File: pkg/jobs/jobspb/jobs.proto (hunk 1 above)
Comment (postamar) [reply pkg/jobs/jobspb/jobs.proto:497]: No, these jobs don't exist yet out in the wild. We'll do a tag ID cleanup pass before the release, this is liable to change again until then.

**Comment by postamar** [comment]
Thanks for the review!

bors r+

**Pull request merged**